src/ruby/end2end/client_memory_usage_client.rb

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env ruby
+
+# Copyright 2018 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative './end2end_common'
+require 'objspace'
+
+def main
+  server_port = ''
+  waiting_time = 5
```

Comment:
Actaully. @Everysick, can you please restructure this test to loop for a certain number of iterations, rather than for a timeout?

E.g., if instead of running `loop do` for a timeout, can we change this to run e.g., `200.times do`? I say 200 because I personally am seeing this bug get triggered > 90% of the time with just 200 iterations, and this will also let the test finish in less than one second. Increasing that number is also ok if you find it necessary.

My thinking is that if the bug is present, than something like 200 iterations should trigger it almost every time and definitely a red flag. But if the bug is not there, then 5 seconds is too much time to wait for.

Reply:
I replaced to a 200 times loop. (rebased)